File: src/main/java/com/blink/backend/controller/configuration/dto/AvailabilityConfigurationDTO.java
package com.blink.backend.controller.configuration.dto;


import com.blink.backend.persistence.entity.appointment.ClinicAvailability;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;

@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class AvailabilityConfigurationDTO {

    private Integer clinicId;
    private String weekDay;
    @JsonFormat(pattern = "HH:mm")
    private LocalTime open;
    @JsonFormat(pattern = "HH:mm")
    private LocalTime close;
    @JsonFormat(pattern = "HH:mm")
    private LocalTime breakStart;
    @JsonFormat(pattern = "HH:mm")
    private LocalTime breakEnd;
    @JsonProperty("is_work_day")
    private Boolean workDay;


    public static AvailabilityConfigurationDTO fromEntity(ClinicAvailability clinicAvailability){
        return AvailabilityConfigurationDTO.builder()
                .weekDay(clinicAvailability.getWeekDay().name())
                .open(clinicAvailability.getOpenTime())
                .close(clinicAvailability.getCloseTime())
                .breakStart(clinicAvailability.getLunchStartTime())
                .breakEnd(clinicAvailability.getLunchEndTime())
                .workDay(clinicAvailability.getIsWorkingDay())
                .build();
    }
}

File: src/main/java/com/blink/backend/controller/configuration/dto/AvailabilityConfigurationDto.kt
package com.blink.backend.controller.configuration.dto

import com.blink.backend.persistence.entity.appointment.ClinicAvailability
import com.fasterxml.jackson.annotation.JsonInclude
import java.time.LocalTime

@JsonInclude(JsonInclude.Include.NON_NULL)
data class AvailabilityConfigurationDto(
    val weekDay: String?,
    val open: LocalTime?,
    val close: LocalTime?,
    val breakStart: LocalTime?,
    val breakEnd: LocalTime?,
    val workDay: Boolean
) {
    init {
        if (workDay) {
            require(open != null) { "open is required when isWorkingDay is true" }
            require(close != null) { "close is required when isWorkingDay is true" }
        }
        require((breakStart == null) == (breakEnd == null)) {
            "breakStart and breakEnd must both be provided or both be null"
        }
    }

    companion object {
        fun fromEntity(clinicAvailability: ClinicAvailability): AvailabilityConfigurationDto {
            return AvailabilityConfigurationDto(
                weekDay = clinicAvailability.weekDay.name,
                open = clinicAvailability.openTime,
                close = clinicAvailability.closeTime,
                breakStart = clinicAvailability.lunchStartTime,
                breakEnd = clinicAvailability.lunchEndTime,
                workDay = clinicAvailability.isWorkingDay
            )
        }
    }
}
